packages/web/src/js/ScrollView.ts

```diff
@@ -0,0 +1,126 @@
+// The `ScrollView` component is currently untested. This is because it is not possible to test it in a headless
+// environment (tests fail on `getBoundingClientRect` that we need to mock somehow).
+
+import BaseComponent from './BaseComponent';
+import { EventHandler, SelectorEngine } from './dom';
+import { debounce, enableToggleAutoloader } from './utils';
+
+const NAME = 'scrollView';
+const DEFAULT_DIRECTION = 'vertical';
+const VIEWPORT_SELECTOR = '[data-spirit-element="viewport"]';
+const CONTENT_SELECTOR = '[data-spirit-element="content"]';
+const IS_SCROLLED_AT_START_CLASS_NAME = 'is-scrolled-at-start';
+const IS_SCROLLED_AT_END_CLASS_NAME = 'is-scrolled-at-end';
+const RESIZE_EVENT = 'resize';
+const SCROLL_EVENT = 'scroll';
+
+// Method `getElementsPositionDifference` sometimes returns floating point values that results in inaccurate detection
+// of start/end. It is necessary to accept this inaccuracy and take every value less or equal to 1px as start/end.
+const EDGE_DETECTION_INACCURACY_PX = 1;
+
+export type Direction = 'horizontal' | 'vertical';
+
+interface ICurrentPosition {
```

Comment:
Please rename according to our conventions from `ICurrentPosition` to `CurrentPositionProps`, thanks

Reply:
Can this interface have more options than what is defined? With that first line, you're actually explaining that any string with a numeric value can be part of an object. I recommend using one or the other option, but not a combination

Reply:
There is only one interface using the `I` prefix and it is in the `BaseComponent`. I suggest not using it and I am ok with `CurrentPosition`.

Also, I do not agree with the `Props` suffix yet, because firstly we are not in React and secondly we do not have discussed any guidelines about that.

Reply:
> Can this interface have more options than what is defined? With that first line, you're actually explaining that any string with a numeric value can be part of an object. I recommend using one or the other option, but not a combination

@adamkudrna What are you trying to say with this interface? That the key can be extended by the developer? Maybe we can suggest a better typing here.

Reply:
I have literally no idea. Please fix it to whatever you feel is the best.

Reply:
Fixed.